src/Rules/PdoStatementExecuteMethodRule.php

```diff
@@ -70,16 +73,33 @@ private function checkErrors(MethodReflection $methodReflection, MethodCall $met
 
         $args = $methodCall->getArgs();
         if (\count($args) < 1) {
-            $parameters = [];
+            $parameterKeys = [];
+            $parameterValues = [];
+
+            $calls = $stmtReflection->findPrepareBindCalls($methodCall);
+
+            foreach ($calls as $bindCall) {
+                $args = $bindCall->getArgs();
+                if (\count($args) >= 2) {
+                    $keyType = $scope->getType($args[0]->value);
+                    if ($keyType instanceof ConstantIntegerType || $keyType instanceof ConstantStringType) {
```

Comment:
do we need this check here, or could we just feed everything into `$parameterKeys` and `resolveParameters` later on would decide which types are supported and which not?

Reply:
It's constrained by the constructor, so phpstan complained.